### PR TITLE
Add print & REPL functionality to CLI

### LINKED ISSUE #142: Create Print() function
Arises from:
https://github.com/jasonwilliams/boa/issues/74

```
1. `print()` function in the global scope (or similar). SpiderMonkey's JavaScript shell defines this as: 
    > Evaluates the expression(s) and displays the result(s) on `stdout`, separated by spaces (" ") and terminated by a newline ("\n").
```

For Boa Shell we need a print function, the implementation is virtually the same as https://github.com/jasonwilliams/boa/blob/master/src/lib/js/console.rs#L11-L98

It might be worth abstracting the above logic into its own function, and have console.log() and print() use it.

print() should only be available in [shell](https://github.com/jasonwilliams/boa/blob/master/src/bin/shell.rs) 

So you will need to add it to the realm there.
You may need to create a helper function in realm which lets you add Objects to the global scope.

#### Spidermonkey equivalent
https://github.com/mozilla/gecko-dev/blob/master/js/src/shell/js.cpp#L2687-L2724

### PR DESCRIPTION
**Pull request by JohnDoneth**
I believe this should close #142 and should make some progress towards #74.

If this PR (gets accepted / is satisfactory) I can continue working on other features needed to complete #74.

### CODE REVIEW

**Comment by jasonwilliams** [comment]
Hey this looks great!

File: boa_cli/src/main.rs [hunk 1]
```diff
@@ -2,29 +2,30 @@
...
 use std::{fs::read_to_string, path::PathBuf};
 use structopt::StructOpt;
-
 /// CLI configuration for Boa.
 #[derive(Debug, StructOpt)]
 #[structopt(author, about)]
 struct Opt {
     /// The javascript file to be evaluated.
-    #[structopt(name = "FILE", parse(from_os_str), default_value = "tests/js/test.js")]
+    #[structopt(name = "FILE", parse(from_os_str))]
```
Comment (jasonwilliams) [inline boa_cli/src/main.rs]: So if i understand it by default boa_cli will now be a shell, and we need to pass the file if we want to keep the same functionality as before?
Im personally OK with that as it matches up with Node etc

File: boa_cli/src/main.rs [hunk 2]
```diff
@@ -2,29 +2,30 @@
...
     files: Vec<PathBuf>,
-    /// Open a boa shell (WIP).
+    /// Open a boa shell.
```
Comment (jasonwilliams) [inline boa_cli/src/main.rs]: Does this option make sense now that shell is the default?

File: boa_cli/src/main.rs (hunk 1 above)
Comment (JohnDoneth) [reply boa_cli/src/main.rs]: Yup! It should be functionally equivalent. I just tested running node locally to check and passing no arguments (just calling `node`) opens an interactive shell.

File: boa_cli/src/main.rs (hunk 2 above)
Comment (JohnDoneth) [reply boa_cli/src/main.rs]: It is now removed!

**Comment by jasonwilliams** [comment]
Do you mind updating the readme and maybe docs/debugging.md so that new users know to pass a file?

Maybe show the argument boa takes or something like https://github.com/ogham/exa/blob/master/README.md#options or https://github.com/sharkdp/fd/blob/master/README.md#command-line-options

**Pull request merged**